src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryInstrumenter.cs

```diff
@@ -88,102 +92,148 @@ static bool GetCommandAndKey(
             StackExchangeRedisConnectionInstrumentation.CreationTags,
             startTime: command.CommandCreated);
 
-        if (activity == null)
+        if (activity is null && metrics.Enabled is false)
         {
             return null;
         }
 
-        activity.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        activity?.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        var meterTags = metrics.Enabled ?
+            new TagList(StackExchangeRedisConnectionInstrumentation.CreationTags) :
+            default(IList<KeyValuePair<string, object?>>);
 
-        if (activity.IsAllDataRequested)
-        {
-            // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md
+        // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md
+
+        // Timing example:
+        // command.CommandCreated; //2019-01-10 22:18:28Z
 
-            // Timing example:
-            // command.CommandCreated; //2019-01-10 22:18:28Z
+        // command.CreationToEnqueued;      // 00:00:32.4571995
+        // command.EnqueuedToSending;       // 00:00:00.0352838
+        // command.SentToResponse;          // 00:00:00.0060586
+        // command.ResponseToCompletion;    // 00:00:00.0002601
 
-            // command.CreationToEnqueued;      // 00:00:32.4571995
-            // command.EnqueuedToSending;       // 00:00:00.0352838
-            // command.SentToResponse;          // 00:00:00.0060586
-            // command.ResponseToCompletion;    // 00:00:00.0002601
+        // Total:
+        // command.ElapsedTime;             // 00:00:32.4988020
 
-            // Total:
-            // command.ElapsedTime;             // 00:00:32.4988020
+        var flags = command.Flags.ToString();
+        activity?.SetTag(SemanticConventions.AttributeDbRedisFlagsKeyName, flags);
+        meterTags?.Add(SemanticConventions.AttributeDbRedisFlagsKeyName, flags);
 
-            activity.SetTag(StackExchangeRedisConnectionInstrumentation.RedisFlagsKeyName, command.Flags.ToString());
+        var operationName = command.Command ?? string.Empty;
+        activity?.SetTag(SemanticConventions.AttributeDbOperationName, operationName);
+        meterTags?.Add(SemanticConventions.AttributeDbOperationName, operationName);
 
+        if (activity is not null)
+        {
             if (options.SetVerboseDatabaseStatements)
             {
                 var (commandAndKey, script) = MessageDataGetter.Value.Invoke(command);
 
                 if (!string.IsNullOrEmpty(commandAndKey) && !string.IsNullOrEmpty(script))
                 {
-                    activity.SetTag(SemanticConventions.AttributeDbStatement, commandAndKey + " " + script);
+                    activity.SetTag(SemanticConventions.AttributeDbQueryText, commandAndKey + " " + script);
```

Comment:
is this intended to be part of this PR? If this is unrelated to adding Metrics, please make it separate PR, so it is easy to review changes.

Reply:
understood, I thought I would give my maximum contribution to this PR but I started to understand the complexity of managing naming, conventions and following guidelines. Without it, it could be a mess.

I removed all changes that applied to activities, in the next PR I will explore the conventions for tracing and their implementation. Sorry for any confusion